conda/petsc/build.sh

```diff
@@ -36,18 +36,19 @@ else
     FCFLAGS="${FCFLAGS} -I$PREFIX/include"
 fi
 
+# Remove std=C++17 from CXXFLAGS as we specify the C++ dialect for PETSc as C++11 in configure_petsc.
+# Specifying both causes an error as of PETSc 3.17, and C++11 as the dialect seems to still be needed
+# for min clang/gcc.
+CXXFLAGS=${CXXFLAGS//-std=c++[0-9][0-9]}
+
 source $PETSC_DIR/configure_petsc.sh
 configure_petsc \
     --COPTFLAGS=-O3 \
     --CXXOPTFLAGS=-O3 \
     --FOPTFLAGS=-O3 \
     --with-x=0 \
     --with-ssl=0 \
-    CC="$CC" \
-    CXX="$CXX" \
-    FC="$FC" \
-    F90="$F90" \
-    F77="$F77" \
+    --with-mpi-dir=$PREFIX \
```

Comment:
How does this work?

Reply:
MPI (via `moose-mpich`) is installed in the `$PREFIX` location as defined by `conda-build`. This is the temporary environment that build creates and then PETSC is also installed to (see `--prefix=$PREFIX` a few lines down in the same file). Then packaging occurs and generalizes this. This only impacts the build and installation - conda takes care of the rest. 

Does that make sense?

Reply:
oh, absolutely. I would have gotten that had I looked at the path... whoops

Reply:
Using $PREFIX here still maintains consistency with the end-user environment. This PETSc will always use the MPI provided by conda.